server/src/main/java/io/deephaven/server/session/SessionServiceGrpcImpl.java

```diff
@@ -290,24 +316,138 @@ public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(final ServerCall<Re
+                final Method[] authMethods = authWiring.getClass().getDeclaredMethods();
+
+                int authMethodIndex = 0;
+                for (; authMethodIndex < authMethods.length; ++authMethodIndex) {
+                    if (authMethods[authMethodIndex].getName().equals(authMethodName)) {
+                        break;
                     }
-                    return new ServerCall.Listener<>() {};
+                }
+
+                if (authMethodIndex == authMethods.length) {
+                    log.error().append("Could not find auth method for service: ").append(serviceName)
+                            .append(" method: ").append(methodName).endl();
+                    return null;
+                }
+
+                return authMethods[authMethodIndex];
+            };
+
+            final boolean isClientStreaming = !call.getMethodDescriptor().getType().clientSendsOneMessage();
+
+            final Method authMethod;
+            if (authWiring == null) {
+                authMethod = null;
+            } else {
+                authMethod = authMethodLookup.get();
+                if (authMethod == null) {
+                    return onFailure.get();
                 }
             }
+
+            // Create a final reference to the auth context for the innerHandler
+            final AuthContext authContext = session == null ? null : session.getAuthContext();
+
+            // On the inner half of the call we'll validate that the request is authorized.
+            final ServerCallHandler<ReqT, RespT> innerHandler = (innerCall, headers) -> {
+                // client streaming calls allow for zero messages, so we check authorization now
+                if (isClientStreaming && authMethod != null) {
+                    try {
+                        authMethod.invoke(authWiring, authContext);
+                    } catch (IllegalAccessException | InvocationTargetException e) {
+                        final Throwable cause = e.getCause();
+                        if (cause instanceof StatusRuntimeException) {
+                            final StatusRuntimeException sre = (StatusRuntimeException) cause;
+                            Status.Code status = sre.getStatus().getCode();
+                            if (status.equals(Status.Code.UNAUTHENTICATED)) {
+                                log.info().append(serviceName).append(".").append(methodName)
+                                        .append(": request unauthenticated").endl();
+                            } else if (status.equals(Status.Code.PERMISSION_DENIED)) {
+                                log.info().append(serviceName).append(".").append(methodName)
+                                        .append(": permission denied ").endl();
+                            } else {
+                                log.error().append(cause).endl();
+                            }
+
+                            call.close(sre.getStatus(), sre.getTrailers());
```

Comment:
Do the same comments you have in the `onFailure` part about catching exceptions for close also make sense here?